internal/controllers/watch/pruning.go

```diff
@@ -0,0 +1,48 @@
+package watch
+
+import (
+	"context"
+	"fmt"
+
+	apiv1 "github.com/Azure/eno/api/v1"
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type pruningController struct {
```

Comment:
Now that we write the input revisions to the composition status (not the current synthesis) we need to clean them up eventually